File: cli/tests/golden/artifacts/typescript/customer.Customer.v1.ts
/**
 * @modelable domain: customer
 * @modelable name: Customer
 * @modelable owner: customer-team
 * @modelable kind: entity
 * @modelable version: 1
 * @modelable changeKind: additive
 */
import type { CatalogProductV1 } from "./catalog.Product.v1";

export interface CustomerCustomerV1 {
  customerId: string;
  displayName: string;
  email: string;
  internalRiskNotes?: string;
  status: 'active' | 'suspended' | 'deleted';
  tags: string[];
  metadata: Record<string, number>;
  address?: { line1: string; line2?: string };
  favoriteProduct?: CatalogProductV1;
  createdAt: string;
  updatedAt?: string;
}
export type Customer = CustomerCustomerV1;

File: cli/tests/golden/artifacts/python/customer/customer_customer_v1.py
from __future__ import annotations

from dataclasses import dataclass
from datetime import date, datetime, time, timedelta
from decimal import Decimal
from typing import Optional
from uuid import UUID

@dataclass(frozen=True, slots=True)
class CustomerCustomerV1:
    customerId: UUID
    displayName: str
    email: str
    status: str
    tags: list[str]
    metadata: dict[str, int]
    createdAt: datetime
    internalRiskNotes: Optional[str] = None
    address: Optional[CustomerCustomerV1Address] = None
    favoriteProduct: Optional[str] = None
    updatedAt: Optional[datetime] = None

@dataclass(frozen=True, slots=True)
class CustomerCustomerV1Address:
    line1: str
    line2: Optional[str] = None
